src/lib/Liblog/chk_file_sec.c

```diff
@@ -452,3 +452,77 @@ tmp_file_sec(char *path, int isdir, int sticky, int disallow, int fullpath)
 	free(real);
 	return (rc);
 }
+
+/**
+ * @brief - This function takes an <program name> <args> as input in string format
+ *	    and returns the program name
+ *
+ * @return - char *
+ * @retval - NULL when no valid program name is found
+ * @retval - a newly allocated program name
+ *
+ * @NOTE - Caller holds the responsibility of freeing up the return value
+ */
+char *
+get_script_name(char *input) {
+	char * tok;
+	char *next_space;
+	int ret_fs;
+	int path_exists = 0;
+	char *prev_space = NULL;
+	int starts_with_quotes = 0;
+	char *delim = " ";
+	struct stat sbuf;
+
+	if (input == NULL)
+		return NULL;
+
+	/* if path starts with double quotes, skip it */
+	if (input[0] == '\"') {
+		input++;
+		starts_with_quotes = 1;
+	}
+
+	tok = strdup(input);
+	if (tok == NULL)
+		return NULL;
+
+	/* get rid of double quotes at the end */
+	if (starts_with_quotes && tok[strlen(tok)-1] == '\"')
+		tok[strlen(tok)-1] = '\0';
+
+	next_space = strpbrk(tok, delim);
+	for (; next_space != NULL; next_space = strpbrk(next_space, delim)) {
+		*next_space = '\0';
+		memset (&sbuf, 0, sizeof(struct stat));
+		ret_fs = stat(tok, &sbuf);
+		if (ret_fs != 0) {
+			if (path_exists == 1) {
+				*prev_space = '\0';
+				return (tok);
+			}
+		}
+		else if (S_ISREG(sbuf.st_mode)) {
+			path_exists = 1;
+			prev_space = next_space;
+		}
+		*next_space = ' ';
+		next_space += strspn(next_space, delim);
+	}
+
+	if (path_exists == 1) {
+		/* set last known space as '\0' so that returned path contains no arguments */
+		*prev_space = '\0';
+		return tok;
+	}
+
+	/* If control is here then it would mean that "tok" must have file path*/
```

Comment:
tok could have had a file path even before control reached here right?

Reply:
I see, so maybe the comment should be "/* If control is here then it would mean that "tok" must have **only** file path*/" ?